HRC_TXING_CHECK/HRC_Txing_Check.py

```diff
@@ -30,58 +50,304 @@
+    
+    #wt_index points to the WT parameter block command in the load.
+    present_cmd_index_load = wt_index
+    
+    # Set the warning flags to False as we want to report only the first
+    # sequence and/or timing warning
+    warning_sequence_flag = False
+    warning_timing_flag = False
+    
+    # Loop through the comands for this SI mode (skipping the parameter
+    # block/bias value line) and,
+    # 1) Check that the previous command matches the expected si
+    #     mode command and
+    # 2) compare the delta t between the present command and the one before it
+
+    # For every command that comes before the WT command.....
+    while present_cmd_index_simode != 0:
+        # Extract the expected command from the si_mode_command_list
+        each_expected_cmd = si_mode_command_list[present_cmd_index_simode-1]
+        
+        # Extract the actual ACIS command from the load line
+        actual_cmd = extract_command_packet(extracted_cmds[present_cmd_index_load - 1])
+    
+        # COMMAND MNEMONIC CHECK - Check to see if the command matches the expected command
+        if (warning_sequence_flag == False) and (each_expected_cmd[0] != actual_cmd):
+                    
+            # WARNING -  Place a warning statement in the comments list, and print it out for the log file.
+            full_comment = " ".join(( ">>> WARNING - ", extracted_cmds[present_cmd_index_load - 1]["date"], "The actual command in the load: ", actual_cmd, "does not match the expected command: ", each_expected_cmd[0] ))
+            
+            # Inform the Load Reviewer in real time that a problem was found.
+            print("\n",extracted_cmds[present_cmd_index_load - 1]["date"], full_comment)
+            
+            # Append the comment to the comment list
+            comment_list.append([extracted_cmds[present_cmd_index_load - 1]["date"], extracted_cmds[present_cmd_index_load - 1]["time"], full_comment])
+            # Set the warning flag to True so that only one warning of this type is given
+            warning_sequence_flag = True
+
+        # TIMING CHECK  - Correct command; now check the time delta between this
+        # command and the one before it in the load
+        else: 
+                # delta between this command and the one before it in the load
+            dt_load = extracted_cmds[present_cmd_index_load]["time"] - extracted_cmds[present_cmd_index_load - 1]["time"]
+
+            # Extract the expected dt between this command and the one before
+            # it in the SI mode command list
+            dt_smcl = each_expected_cmd[1]           
+            
+            # If the time pointed to by present_cmd_index minus dt is equal
+            # to the time of the previous command, all is well.
+            if (warning_timing_flag == False) and (not math.isclose( dt_load, dt_smcl, rel_tol = epsilon)):
```

Comment:
I couldn't find `rel_tol` in this diff. where is it?

Reply:
Sorry, my mistake

Reply:
rel_tol is a parameter to math.isclose